src/client.js

```diff
@@ -23,12 +23,29 @@ const API_TYPE_CLIENT = 'client';
 const API_TYPE_APP = 'app';
 
 /**
- * Create a new StitchClient instance.
+  * Factory class to create a new StitchClient asynchronously.
+  */
+export class StitchClient {
+  constructor() {
+    throw new StitchError('Constructor undefined. Use `StitchClient.init` instead.');
+  }
+
+  static init(clientAppID, options = {}) {
+    return new Promise((resolve, reject) => {
+      resolve(new _StitchClient(clientAppID, options));
+    });
+  }
+}
+
+/**
+ * Create a new _StitchClient instance.
+ * This is the internal implementation for StitchClient and should not
+ * be exposed.
  *
  * @class
  * @return {StitchClient} a StitchClient instance.
  */
-export default class StitchClient {
+export class _StitchClient {
   constructor(clientAppID, options = {}) {
```

Comment:
This is probably what should become the private function and not the actual constructor. This can return a promise itself and not have to hold any vars regarding initialization like `_isInitialized` and `_isInitialized`.